client/state/stored-cards/reducer.js

```diff
@@ -26,25 +28,14 @@ import {
  * @param  {Object} action storeCard action
  * @return {Array}         Updated state
  */
-export const items = ( state = [], action ) => {
-	switch ( action.type ) {
-		case STORED_CARDS_ADD_COMPLETED:
-			return [ ...state, action.item ];
-
-		case STORED_CARDS_FETCH_COMPLETED:
-			return action.list;
+export const items = createReducer( [], {
+	[ STORED_CARDS_ADD_COMPLETED ]: ( state, action ) => [ ...state, action.item ],
 
-		case STORED_CARDS_DELETE_COMPLETED:
-			return state.filter( item => item.stored_details_id !== action.card.stored_details_id );
+	[ STORED_CARDS_FETCH_COMPLETED ]: ( state, action ) => action.list,
```

Comment:
Those action handlers look very good. You can also consider using ES6 destructuring like:
```js
( state, { item } ) => [ ...state, item ]
( state, { list } ) => list
```
Totally up to you :)